app/views/devise/two_factor_authentication_setup/index.html.slim

```diff
@@ -8,6 +8,6 @@
       = simple_form_for(resource, as: :user, html: { autocomplete: 'off', role: 'form' }, url: users_otp_path) do |f|
         = f.error_notification
         p = t('devise.two_factor_authentication.otp_setup')
-        = f.input :mobile, label: 'Mobile Number', required: true, autofocus: true
+        = f.input :mobile, autofocus: true, label: 'Mobile Number', required: true, input_html: { type: :tel }
         = f.button :submit, 'Submit'
```

Comment:
Alternatively, you can replace `input_html: { type: :tel }` with `as: :tel` after `:mobile`. See: https://github.com/plataformatec/simple_form#available-input-types-and-defaults-for-each-column-type